language/move-core/types/src/u256.rs

```diff
@@ -0,0 +1,376 @@
+// Copyright (c) The Move Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::anyhow;
+use ethnum::U256 as EthnumU256;
+use num::{bigint::Sign, BigInt};
+#[cfg(any(test, feature = "fuzzing"))]
+use proptest::strategy::BoxedStrategy;
+#[cfg(any(test, feature = "fuzzing"))]
+use proptest::strategy::Strategy;
+use std::{
+    fmt,
+    mem::size_of,
+    ops::{BitAnd, BitOr, BitXor, Shl, Shr},
+};
+use uint::FromStrRadixErr;
+
+const NUM_BITS_PER_BYTE: usize = 8;
+const U256_NUM_BITS: usize = 256;
+const U256_NUM_BYTES: usize = U256_NUM_BITS / NUM_BITS_PER_BYTE;
+pub type U256FromStrError = FromStrRadixErr;
```

Comment:
You'll probably want to wrap this in a NewType so as to not leak the dep on uint